cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java

```diff
@@ -299,6 +299,53 @@ public Response remove(String id, String rev) {
     }
 
     /**
+     * Reads an attachment from the database.
+     *
+     * The stream must be closed after usage, otherwise http connection leaks will occur.
+     *
+     * @param docId the document id
+     * @param attachmentId the attachment id
+     * @return the attachment in the form of an {@code InputStream}.
+     */
+    public InputStream getAttachment(String docId, String attachmentId) {
+        final URI uri = new DatabaseURIHelper(dbUri).attachmentUri(docId, attachmentId);
+        return getAttachment(uri);
+    }
+
+    /**
+     * Reads an attachment from the database.
+     *
+     * The stream must be closed after usage, otherwise http connection leaks will occur.
+     *
+     * @param docId the document id
+     * @param attachmentId the attachment name
+     * @param revId the document revision id
+     * @return the attachment in the form of an {@code InputStream}.
+     */
+    public InputStream getAttachment(String docId, String attachmentId, String revId) {
+        final URI uri = new DatabaseURIHelper(dbUri).attachmentUri(docId, revId, attachmentId);
+        return getAttachment(uri);
+    }
+
+    /**
+     * Reads an attachment from the database.
+     *
+     * The stream must be closed after usage, otherwise http connection leaks will occur.
+     *
+     * @param uri the attachment URI
+     * @return the attachment in the form of an {@code InputStream}.
+     */
+    private InputStream getAttachment(URI uri) {
+        HttpConnection connection = Http.GET(uri);
+        couchDbClient.execute(connection);
+        try {
+            return connection.responseAsInputStream();
```

Comment:
The `Content-Type` header in particular provides information that could potentially be essential for the attachment stream to be handled correctly by an application, but that information is available from the document itself.